mindmeld/active_learning/classifiers.py

```diff
@@ -0,0 +1,418 @@
+                filtered_unsampled_queries,
+            ) = DataBucket.filter_queries(unsampled_queries, domain)
+            _, filtered_test_queries = DataBucket.filter_queries(test_queries, domain)
+            # Train
+            ic = self.nlp.domains[domain].intent_classifier
+            ic.fit(queries=filtered_sampled_queries)
+            # Evaluate Test Queries
+            ic_eval_test = ic.evaluate(queries=filtered_test_queries)
+            if not ic_eval_test:
+                unsampled_idx_preds_pairs.extend(
+                    (i, self._get_zero_prob_vector())
+                    for i in filtered_unsampled_queries_indices
+                )
+                continue
+            ic_eval_test_dict[domain] = ic_eval_test
+            # Get Probability Vectors
+            ic_queries_prob_vectors = MindMeldALClassifier._get_probs(
+                classifier=ic,
+                queries=filtered_unsampled_queries,
+                nlp_component_to_id=intent2id[domain],
+            )
+            intents = [
+                f"{domain}|{intent}"
+                for intent in ic_eval_test.get_stats()["class_labels"]
+            ]
+            padded_ic_queries_prob_vectors = self._pad_intent_probs(
+                ic_queries_prob_vectors, intents
+            )
+            for i in range(len(filtered_unsampled_queries)):
+                unsampled_idx_preds_pairs.append(
+                    (
+                        filtered_unsampled_queries_indices[i],
+                        padded_ic_queries_prob_vectors[i],
+                    )
+                )
+        unsampled_idx_preds_pairs.sort(key=lambda x: x[0])
+        padded_ic_queries_prob_vectors = [x[1] for x in unsampled_idx_preds_pairs]
+        return padded_ic_queries_prob_vectors, ic_eval_test_dict
+
+    def _get_zero_prob_vector(self):
+        """
+        Returns:
+            zero_prob_vector (List): Array of zeroes with a length equal to the total number of
+                intents.
+        """
+        return [0] * len(self.intent2idx)
+
+    @staticmethod
+    def _update_eval_stats_intent_level(
+        eval_stats: defaultdict, ic_eval_test_dict: Dict
+    ):
+        """Update the eval_stats dictionary with evaluation metrics from intent
+        classifiers.
+
+        Args:
+            eval_stats (defaultdict): Evaluation metrics to be included in accuracies.json
+            ic_eval_test_dict (Dict): Dictionary mapping a domain (str) to the
+                associated ic_eval_test object.
+        """
+        for domain, ic_eval_test in ic_eval_test_dict.items():
+            eval_stats["accuracies"][domain] = {
+                "overall": ic_eval_test.get_stats()["stats_overall"]["accuracy"]
+            }
+            for i, intent in enumerate(ic_eval_test.get_stats()["class_labels"]):
+                eval_stats["accuracies"][domain][intent] = {
+                    "overall": ic_eval_test.get_stats()["class_stats"]["f_beta"][i]
```

Comment:
Is f_beta the only reportable metric? Maybe we can configure this such that someone can specify other types as well